### PR TITLE
patch: Create balena-ci.yml

### PR DESCRIPTION
**Pull request by vipulgupta2048**
@rcooke-warwick  Adding the API key of the testbot user account.

### CODE REVIEW

**Comment by rcooke-warwick** [comment]
@vipulgupta2048 does this set things up so we can automatically deploy a leviathan release to a fleet?

**Comment by vipulgupta2048** [comment]
Very much indeed, the good life awaits.

**Comment by klutchell** [comment]
@vipulgupta2048 I added the testbot fleet instead of the example, and I pinned the action versions to specific releases.

Dependabot can bump the actions versions, but we need to add a balenaCloud token to our secrets. We should check how @ab77 did it for balena-jenkins and similar to avoid using his own.

File: .github/dependabot.yml [hunk 1]
```diff
@@ -1,10 +1,17 @@
...
+      interval: weekly
+    open-pull-requests-limit: 0
+    commit-message:
+      prefix: patch
+    rebase-strategy: disabled
+  - package-ecosystem: github-actions
+    directory: '/'
+    schedule:
+      interval: weekly
+    open-pull-requests-limit: 0
```
Comment (vipulgupta2048) [inline .github/dependabot.yml:15]: @klutchell Is this to keep the GitHub Actions updated?

**Comment by vipulgupta2048** [comment]
@klutchell If I understand this correctly, I think we need to add the secrets to the GitHub repo using admin access. (I have admin access). What I need to ask is, why do we need a GitHub token now?

**Comment by vipulgupta2048** [comment]
I would also say we change the name balena.yml file, as we have already have enough types `balena.yml` files to confuse people.

**Comment by vipulgupta2048** [comment]
@balena-ci rebase

**Comment by klutchell** [comment]
@vipulgupta2048 the GH token is required for the `create_ref` function but `secrets.GITHUB_TOKEN` is available by default so no need to add it as a secret (GH won't let you anyway).

**Comment by vipulgupta2048** [comment]
>  We should check how ab77 did it for balena-jenkins and similar to avoid using his own.

@klutchell Right on right on. For Jenkins, IIRC we have a user called testbot in balenaCloud which provides it's balenaCloud keys to Jenkins. What I will do is, create a new set of keys from that account and add it to this repository. cc: @rcooke-warwick

**Comment by vipulgupta2048** [comment]
@balena-ci rebase

**Comment by vipulgupta2048** [comment]
@balena-ci I self-ceritfy!

**Comment by vipulgupta2048** [comment]
@balena-ci I self-certify!

**Pull request merged**